lib/modules/blockchain_connector/index.js

```diff
@@ -232,9 +233,29 @@ class BlockchainConnector {
   }
 
   deployContractFromObject(deployContractObject, params, cb) {
+    let deployer = this.deployers[deployContractObject._deployData];
+
+    if (!deployer) {
+      deployer = {running: false, requested: []};
+      this.deployers[deployContractObject._deployData] = deployer;
+    }
+
+    if (deployer.running) {
+      deployer.requested.push({deployContractObject, params, cb});
+      return;
+    }
+
+    deployer.running = true;
     embarkJsUtils.secureSend(this.web3, deployContractObject, {
```

Comment:
With the changes in this PR, still getting promise related errors:

```
UnhandledPromiseRejectionWarning: Error: Returned error: replacement transaction underpriced
```
```
UnhandledPromiseRejectionWarning: Error: Returned error: nonce too low
```

Looking at:  https://github.com/embark-framework/EmbarkJS/blob/master/src/utils.js#L27

`secureSend` is obviously a bit complex. I'm wondering whether in the EmbarkJS line linked above  the `cb`, `resolve`, and `reject` aren't quite wired up in the correct way. Maybe a callback shouldn't be involved and callers of `secureSend` should be working with the returned promise?

Reply:
How do you get this error?
Could you describe your flow?

Reply:
@PascalPrecht definitely using the PR branch. @alaibe and I discussed yesterday, and the theory was that remaining errors I was seeing had to do with ENS contracts, which shouldn't be deployed to infura to begin with.